script-test/playbackstrategies/msestrategytest.js

```diff
@@ -1,579 +1,525 @@
 
-          mockVideoElement.addEventListener.and.callFake(function (eventType, handler) {
-            eventHandlers[eventType] = handler;
+        mockDashInstance.duration.and.returnValue(101);
 
-            eventCallbacks = function (event) {
-              eventHandlers[event].call(event);
-            };
-          });
+        mockDashjs.MediaPlayer.and.returnValue(mockDashMediaPlayer);
+        mockDashMediaPlayer.create.and.returnValue(mockDashInstance);
 
-          // For DVRInfo Based Seekable Range
-          mockDashInstance.isReady.and.returnValue(true);
-          mockDashInstance.getDashMetrics.and.returnValue({
-            getCurrentDVRInfo: function () {
-              return {range: {
-                start: 0,
-                end: 101
-              }};
-            }
-          });
-
-          window.dashjs = mockDashjs;
-
-          spyOn(document, 'createElement').and.returnValue(mockVideoElement);
-          mockDashInstance.getDebug.and.returnValue(mockDashDebug);
-        }
+        mockMediaElement.addEventListener.and.callFake(function (eventType, handler) {
+          eventHandlers[eventType] = handler;
 
-        afterEach(function () {
-          mockVideoElement.currentTime = 0;
-          document.body.removeChild(videoContainer);
+          eventCallbacks = function (event) {
+            eventHandlers[event].call(event);
+          };
         });
 
-        describe('Transitions', function () {
-          it('canBePaused() Transition is true', function () {
-            setUpMSE();
-
-            expect(mseStrategy.transitions.canBePaused()).toBe(true);
-          });
-
-          it('should create an audio element and add it to the video container', function () {
-            setUpMSE(null, null, MediaKinds.AUDIO);
-
-            expect(videoContainer.childElementCount).toBe(0);
-
-            mseStrategy.load('src', null, 0);
-
-            expect(videoContainer.firstChild).toBe(mockVideoElement);
-            expect(videoContainer.childElementCount).toBe(1);
-          });
-
-          it('canBeginSeek() Transition is true', function () {
-            setUpMSE();
-
-            expect(mseStrategy.transitions.canBeginSeek()).toBe(true);
-          });
+          // For DVRInfo Based Seekable Range
+        mockDashInstance.isReady.and.returnValue(true);
+        mockDashInstance.getDashMetrics.and.returnValue({
+          getCurrentDVRInfo: function () {
+            return {range: {
+              start: 0,
+              end: 101
+            }};
+          }
         });
 
-        describe('Load', function () {
-          it('should create a video element and add it to the video container', function () {
-            setUpMSE();
+        window.dashjs = mockDashjs;
 
-            expect(videoContainer.childElementCount).toBe(0);
+        spyOn(document, 'createElement').and.returnValue(mockMediaElement);
```

Comment:
Sorry @JaKraken, this is the crucial bit that needs to change to test the correct thing. As it is, calling `createElement` with anything will give you what you're looking for, so the test will always pass. You should be able to see this by reverting your implementation and observing that your test erroneously passes (TDD would've helped you here 😛).

You need to change this mock to behave differently depending on whether it is called with `'video'` or `'audio'` to test the functionality correctly. This can be done with `and.callFake` rather than `and.returnValue`. See https://jasmine.github.io/2.0/introduction.html#section-Spies:_%3Ccode%3Eand.callFake%3C/code%3E.